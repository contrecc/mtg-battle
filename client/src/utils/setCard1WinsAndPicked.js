export default function setCard1WinsAndPicked() {
  return function(state, props) {
    return {
      winner: true,
      winStreak: state.winStreak + 1,
      winningCard: 'card1',
      roundCompleted: true
    };
  };
}
