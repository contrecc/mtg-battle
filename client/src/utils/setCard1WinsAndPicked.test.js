import setCard1WinsAndPicked from './setCard1WinsAndPicked';

describe('setCard1WinsAndPicked', () => {
  it('should return a function', () => {
    expect(setCard1WinsAndPicked()).toMatchSnapshot();
  });

  it('should return a function that then updates the state', () => {
    const returnedFunction = setCard1WinsAndPicked();
    expect(returnedFunction({winStreak: 1}, {})).toMatchSnapshot();
  });
});
