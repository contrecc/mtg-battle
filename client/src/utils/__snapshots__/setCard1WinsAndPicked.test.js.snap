// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setCard1WinsAndPicked should return a function 1`] = `[Function]`;

exports[`setCard1WinsAndPicked should return a function that then updates the state 1`] = `
Object {
  "roundCompleted": true,
  "winStreak": 2,
  "winner": true,
  "winningCard": "card1",
}
`;
